language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -398,7 +400,14 @@ public static DiagBasedPositionDetails computePositionDetails(SyntaxTree syntaxT
             }
             List<String> memberTypes = new ArrayList<>();
             for (TypeSymbol tMember : tMembers) {
-                memberTypes.add(CodeActionUtil.getPossibleType(tMember, edits, context).orElseThrow());
+                String possibleType = CodeActionUtil.getPossibleType(tMember, edits, context).orElseThrow();
+                if (tMember.typeKind() == TypeDescKind.UNION || tMember.typeKind() == TypeDescKind.SINGLETON) {
```

Comment:
For what sort of scenarios would the `tMember` be a union type? 🤔

Reply:
```
type t t1|t2;
type t1 "a"|"b"|"c";
type t2 "p"|"q"|"r";
```
If we are looking at `t`, I we get `"a"|"b"|"c"` as a union type. @TharushiJay can you confirm?

Reply:
Consider the scenario, 
type t "0"|"1"|t1;
type t1 "a"|"b"|"c";

Reply:
Consider the scenario,
```
type t "0"|"1"|t1;
type t1 "a"|"b"|"c";
```
<img width="486" alt="Screenshot 2021-09-02 at 20 40 52" src="https://user-images.githubusercontent.com/27485094/131869779-52002a3c-a484-41cd-a81a-95835d223b41.png">

Here, we don't get `t` in `tMembers`. Instead we get `"0"|"1"` as a union type. That is, when the type definition has both, more than 1 terminal and at least 1 non-terminal.

Reply:
Hmm. That's an issue. This should ideally have been a union type containing 3 members: 2 singleton types and a type ref. Will check this and see.

Reply:
If that's the case, we need to get that fixed. @TharushiJay shall we create an issue?